catalog/dags/database/catalog_cleaner/catalog_cleaner.py

```diff
@@ -0,0 +1,219 @@
+"""
+Catalog Data Cleaner DAG
+
+Use the TSV files created during the cleaning step of the ingestion process to bring
+the changes into the catalog database and make the updates permanent.
+
+The DAG has a structure similar to the batched_update DAG, but with a few key
+differences:
+  1. Given the structure of the TSV, it updates a single column at a time.
+  2. The batch updates are parallelized to speed up the process. The maximum number of
+     active tasks is limited to 3 (at first to try it out and) to avoid overwhelming
+     the database.
+  3. It needs slightly different SQL queries to update the data. One change for example,
+     is that it only works with the `image` table given that is the only one where the
+     cleaning steps are applied to in the ingestion server.
+"""
+
+import logging
+from datetime import timedelta
+
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.models.abstractoperator import AbstractOperator
+from airflow.models.param import Param
+from airflow.operators.python import get_current_context
+from airflow.utils.trigger_rule import TriggerRule
+
+from common import slack
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID
+from common.sql import (
+    RETURN_ROW_COUNT,
+    PGExecuteQueryOperator,
+    PostgresHook,
+    single_value,
+)
+from database.batched_update.batched_update import run_sql
+from database.catalog_cleaner import constants
+
+
+logger = logging.getLogger(__name__)
+
+
+@task
+def count_dirty_rows_and_notify(temp_table_name: str, task: AbstractOperator = None):
+    """Get the number of rows in the temp table before the updates."""
+    count = run_sql.function(
+        dry_run=False,
+        sql_template=f"SELECT COUNT(*) FROM {temp_table_name}",
+        query_id=f"{temp_table_name}_count",
+        handler=single_value,
+        task=task,
+    )
+    notify_slack.function(
+        text=f"Starting the cleaning process in upstream DB. Expecting {count:,} rows"
```

Comment:
Do we still call the DB upstream, or is it `catalog` DB (vs the API db) ?

Reply:
Would it be possible to add the field being cleaned (`url`, `foreign_landing_url` etc) to this notification?

Reply:
> Do we still call the DB upstream, or is it `catalog` DB (vs the API db) ?

That's a good question. I use both terms intermittently, which can be confusing for sure 😅  Which one do you prefer?

> Would it be possible to add the field being cleaned (url, foreign_landing_url etc) to this notification?

It's part of the temporary table name but thinking again, it's better to make it explicit. Done! :)

Reply:
> Which one do you prefer?
I would prefer `catalog`.
Although you could say that the API is also a "catalog", but I guess I look at it more from the point of view of our stacks. On another hand, since this database (upstream/catalog) is the single source of truth that contains all information which can later be filtered in the API database, it makes sense to consider it "**the** catalog"